lib/kernel/src/socket.erl

```diff
@@ -5553,7 +5538,10 @@ With argument `TimeoutOrHandle`; equivalent to
 `TimeoutOrHandle :: Handle` has been allowed *since OTP 24.0*.
 """.
 -spec recvmsg(Socket :: socket(), Flags :: list()) -> dynamic();
-             (Socket :: socket(), TimeoutOrHandle :: reference() | 'infinity' | 'nowait' | non_neg_integer()) -> dynamic().
+             (Socket :: socket(), TimeoutOrHandle) -> dynamic() when
+      TimeoutOrHandle :: Timeout | Handle,
+      Handle :: select_handle() | completion_handle(),
+      Timeout :: timeout() | 'nowait'.
```

Comment:
I think this should be:
``` erlang
TimeoutOrHandle :: timeout() | `nowait` | Handle
Handle          :: select_handle() | completion_handle()
```
or
``` erlang
TimeoutOrHandle :: timeout() | `nowait` | select_handle() | completion_handle()
```

`Timeout` is just `timeout()`, not `'nowait'`.

`Handle` is `select_handle() | completion_handle()`, not `'nowait'`

`'nowait'` should be in single quotes, for convention.  That has been missed in a few places above

`TimeoutOrHandle` should really be named `TimeoutOrNowaitOrHandle`, since `'nowait'` is neither a `timeout()` nor a `Handle`, but we let that one slip to make the variable name more readable.